test/inductor/test_torchinductor.py

```diff
@@ -13112,6 +13112,192 @@ def wrapper(inp, weight):
             _, code = run_and_get_code(wrapper, inp, weight)
             self.assertTrue("in_out_ptr" in code[1])
 
+        @torch._functorch.config.patch("donated_buffer", True)
+        def test_donated_buffer_inplace_gpt(self):
+            # model implementation from llm.c:
+            # https://github.com/karpathy/llm.c/blob/master/train_gpt2.py
+            class NewGELU(nn.Module):
```

Comment:
Would be great if we could have a more minimal test here. Such as by being able to specify donated buffer for forward (which is generically useful). Any number of possible changes could make this no longer trigger. Potentially either by monkeypatching `get_donated_idxs`, or, making a public api - see this pr : https://github.com/pytorch/pytorch/commit/e0d6072f692ac55508cd4187548b389afc794856. 


At the very least - you need to force pad mm for this test to fire.